src/main/java/wooteco/subway/repository/LineDao.java

```diff
@@ -0,0 +1,66 @@
+package wooteco.subway.repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import wooteco.subway.domain.line.Line;
+
+@Repository
+public class LineDao {
+
+    private static final RowMapper<Line> ROW_MAPPER = (resultSet, rowNumber) -> {
+        long id = resultSet.getLong("id");
+        String name = resultSet.getString("name");
+        String color = resultSet.getString("color");
+        return new Line(id, name, color);
+    };
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public LineDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public long save(Line line) {
+        String saveQuery = "INSERT INTO LINE (name, color) VALUES (?, ?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        PreparedStatementCreator preparedStatementCreator = (connection) -> {
+            PreparedStatement prepareStatement = connection
+                .prepareStatement(saveQuery, new String[]{"id"});
+            prepareStatement.setString(1, line.getName());
+            prepareStatement.setString(2, line.getColor());
+            return prepareStatement;
+        };
+        jdbcTemplate.update(preparedStatementCreator, keyHolder);
+        return keyHolder.getKey().longValue();
+    }
+
+    public List<Line> findAll() {
+        String query = "SELECT * FROM LINE";
+        return jdbcTemplate.query(query, ROW_MAPPER);
+    }
+
+    public Line findById(Long id) {
+        String query = "SELECT * FROM LINE WHERE id = ?";
+        return jdbcTemplate.queryForObject(query, ROW_MAPPER, id);
+    }
+
+    public void updateLine(Long id, String name, String color) {
+        String query = "UPDATE line SET name = ?, color = ? WHERE id = ?";
+        int rowCounts = jdbcTemplate.update(query, name, color, id);
+        if (rowCounts == 0) {
+            throw new EmptyResultDataAccessException(0);
```

Comment:
`jdbcTemplate.update(query, name, color, id);`가 반환하는 값은 무엇일까요? 🤔

```java
	/**
	 * Issue a single SQL update operation (such as an insert, update or delete statement).
	 * @param sql static SQL to execute
	 * @return the number of rows affected
	 * @throws DataAccessException if there is any problem.
	 */
	int update(String sql) throws DataAccessException;
```

그리고 EmptyResultDataAccessException는 어떤 상황에서 사용하는 예외일까요?
```java
/**
 * Data access exception thrown when a result was expected to have at least
 * one row (or element) but zero rows (or elements) were actually returned.
 *
 * @author Juergen Hoeller
 * @since 2.0
 * @see IncorrectResultSizeDataAccessException
 */
@SuppressWarnings("serial")
public class EmptyResultDataAccessException extends IncorrectResultSizeDataAccessException {
```


update가 반환하는 값은 row가 아니기때문에 EmptyResultDataAccessException을 사용할 대상이 아니랍니다.
```java
if (rowCounts == 0)
```
로 갱신된 값이 하나도 없을 떄라는 의미로는 사용 가능하지만 하나도 조회되지 않았을 때 사용하는 EmptyResultDataAccessException과는 의미가 맞지 않습니다.